### DISCUSSION TITLE
Question: replace traefik with thruster?

### DISCUSSION THREAD

**Discussion by awellnitz-materna** [basecamp/kamal#727] [Q&A]
Are there plans to replace traefik with thruster? Personally, I'm not a big fan of Traefik, so the question of how I can replace it quickly arises and since you have a reverse proxy yourself, the question is obvious.

**Comment by brenogazzola** [comment]
Doubt it. They are used for different things. Traeffik is not being used as a reverse proxy in Kamal, but to allow zero downtime deploys.

**Reply by morgoth** [reply]
@awellnitz-materna Looks like Traefik will be replaced with custom tool. Vide https://github.com/basecamp/kamal/pull/500#issuecomment-2029328576

**Comment by nickhammond** [comment]
Is there a PR for this anywhere? I'm a little concerned about a proxy server being built from scratch instead of leveraging an existing option. If you look at Thruster's recent commits it's sort of catching up to standard proxy server functionality([X-Forwarded-For](https://github.com/basecamp/thruster/commit/ec8c5f6b9b06425dbffc4edb43361e48f0afe8da), [X-Sendfilesending back correct Content-Length](https://github.com/basecamp/thruster/commit/3738cdd2328886504bb2045afd3fd7885b81df16)).

**Reply by tannakartikey** [reply]
It might be replaced with kamal-proxy but the source code is not open as yet. 

https://github.com/basecamp/kamal/tree/kamal-proxy

**Reply by nickhammond** [reply]
Ah, good find. Looks like it came from the internal parachute project, renamed to `basecamp/kamal-proxy:latest`
